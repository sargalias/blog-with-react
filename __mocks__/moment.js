import moment from 'moment';

const mockMoment = (date=0) => moment(date);

export default mockMoment;
