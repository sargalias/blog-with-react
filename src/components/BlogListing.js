import React from 'react';

const BlogListing = (props) => (
  <article className="blog-listing">
    <div className="header">
      <div className="aspect-ratio-img">
        <a href="/sdf">
          <img src="https://placeimg.com/500/250/any" height="300" alt="" />
        </a>
      </div>
      <div className="category-text-container"><a href="#" className="btn btn-primary">Travel</a></div>
    </div>
    <div className="content">
      <h3 className="post-title"><a href="sdf">Post title</a></h3>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aspernatur, dolor facere fugit ipsam maxime sequi
        soluta. Accusamus atque ea eaque harum, id impedit itaque magni odit optio quod? Non, officia.</p>
      <a href="#">Read more &raquo;</a>
    </div>
    <div className="metadata-container">
      <div className="metadata">Published by <a href="#">User</a> at <a href="">11/08/2018</a></div>
    </div>
  </article>
);

export default BlogListing;
