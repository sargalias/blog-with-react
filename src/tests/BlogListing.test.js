import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import BlogListing from '../components/BlogListing';

const sampleBlogPost = {
  imageURL: 'https://via.placeholder.com/350x350',
  category: 'Food',
  title: 'Test title',
  excerpt: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Est fugiat harum obcaecati pariatur sint tempore unde. Aperiam commodi distinctio earum eligendi, enim itaque laudantium magni neque porro temporibus vitae voluptatum?',
  postLink: '/test-link',
  userName: 'Test username',
  userLink: 'Test userlink',
  date: moment(0)
};

test('BlogListing should render correctly with props', () => {
  const wrapper = shallow(<BlogListing blogPost={sampleBlogPost} />);
  expect(wrapper).toMatchSnapshot();
});

test('BlogListing should render empty with no props', () => {
  const wrapper = shallow(<BlogListing />);
  expect(wrapper).toMatchSnapshot();
});

test('BlogListing should be empty with no props', () => {
  const wrapper = shallow(<BlogListing />);
  expect(wrapper.html()).toBeNull();
});

test('BlogListing should have the correct post title', () => {
  const wrapper = shallow(<BlogListing blogPost={sampleBlogPost} />);
  expect(wrapper.find('h3').text()).toBe(sampleBlogPost.title);
});

test('BlogListing should have the correct image', () => {
  const wrapper = shallow(<BlogListing blogPost={sampleBlogPost} />);
  const re = new RegExp(`src="${sampleBlogPost.imageURL}"`);
  expect(wrapper.find('img').html()).toMatch(re);
});

test('BlogListing should have the correct excerpt', () => {
  const wrapper = shallow(<BlogListing blogPost={sampleBlogPost} />);
  expect(wrapper.find('p').text()).toBe(sampleBlogPost.excerpt);
});

test('BlogListing excerpt section should have no HTML with no excerpt', () => {
  const blogPost = {
    ...sampleBlogPost,
    excerpt: ''
  };
  const wrapper = shallow(<BlogListing blogPost={blogPost} />);
  expect(wrapper.find('.content #excerpt').length).toBe(0);
});
