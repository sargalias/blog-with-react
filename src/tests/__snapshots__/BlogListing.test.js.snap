// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogListing should render correctly with props 1`] = `
<article
  className="blog-listing"
>
  <div
    className="header"
  >
    <div
      className="aspect-ratio-img"
    >
      <a
        href="/sdf"
      >
        <img
          alt=""
          height="300"
          src="https://via.placeholder.com/350x350"
        />
      </a>
    </div>
    <div
      className="category-text-container"
    >
      <a
        className="btn btn-primary"
        href="#"
      >
        Travel
      </a>
    </div>
  </div>
  <div
    className="content"
  >
    <h3
      className="post-title"
    >
      <a
        href="sdf"
      >
        Test title
      </a>
    </h3>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aspernatur, dolor facere fugit ipsam maxime sequi soluta. Accusamus atque ea eaque harum, id impedit itaque magni odit optio quod? Non, officia.
    </p>
    <a
      href="#"
    >
      Read more »
    </a>
  </div>
  <div
    className="metadata-container"
  >
    <div
      className="metadata"
    >
      Published by 
      <a
        href="#"
      >
        User
      </a>
       at 
      <a
        href=""
      >
        11/08/2018
      </a>
    </div>
  </div>
</article>
`;

exports[`BlogListing should render empty with no props 1`] = `
<article
  className="blog-listing"
>
  <div
    className="header"
  >
    <div
      className="aspect-ratio-img"
    >
      <a
        href="/sdf"
      >
        <img
          alt=""
          height="300"
          src="https://placeimg.com/500/250/any"
        />
      </a>
    </div>
    <div
      className="category-text-container"
    >
      <a
        className="btn btn-primary"
        href="#"
      >
        Travel
      </a>
    </div>
  </div>
  <div
    className="content"
  >
    <h3
      className="post-title"
    >
      <a
        href="sdf"
      >
        Post title
      </a>
    </h3>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aspernatur, dolor facere fugit ipsam maxime sequi soluta. Accusamus atque ea eaque harum, id impedit itaque magni odit optio quod? Non, officia.
    </p>
    <a
      href="#"
    >
      Read more »
    </a>
  </div>
  <div
    className="metadata-container"
  >
    <div
      className="metadata"
    >
      Published by 
      <a
        href="#"
      >
        User
      </a>
       at 
      <a
        href=""
      >
        11/08/2018
      </a>
    </div>
  </div>
</article>
`;
